lib/_http_server.js

```diff
@@ -151,38 +151,32 @@ ServerResponse.prototype._implicitHeader = function() {
   this.writeHead(this.statusCode);
 };
 
-ServerResponse.prototype.writeHead = function(statusCode, reason, obj) {
-  var headers;
-
-  if (typeof reason === 'string') {
-    // writeHead(statusCode, reasonPhrase[, headers])
-    this.statusMessage = reason;
+ServerResponse.prototype.writeHead = function(statusCode, statusMsg, headers) {
+  if (typeof statusMsg === 'string') {
+    // writeHead(statusCode, statusMessage[, headers])
+    this.statusMessage = statusMsg;
   } else {
     // writeHead(statusCode[, headers])
     this.statusMessage =
         this.statusMessage || STATUS_CODES[statusCode] || 'unknown';
-    obj = reason;
+    headers = statusMsg;
   }
   this.statusCode = statusCode;
 
   if (this._headers) {
     // Slow-case: when progressive API and header fields are passed.
-    if (obj) {
-      var keys = Object.keys(obj);
+    if (headers) {
+      var keys = Object.keys(headers);
       for (var i = 0; i < keys.length; i++) {
-        var k = keys[i];
-        if (k) this.setHeader(k, obj[k]);
+        var key = keys[i];
+        this.setHeader(key, headers[key]);
```

Comment:
No such PR seems to exist on joyent/node..

I'd say we should probably keep it in there for now unless we can show it isn't necessary.